Source/Scene/Billboard.js

```diff
@@ -762,6 +809,103 @@ define([
         return this._pickId;
     };
 
+    var scratchRay = new Ray();
+    var scratchPosition = new Cartesian3();
+    var scratchCartographic = new Cartographic();
+
+    Billboard._clampPosition = function(object, tile, mode, projection) {
+        var modeChanged = object._mode !== mode;
+        var ellipsoid = projection.ellipsoid;
+        var level = tile.level;
+
+        if (level > object._level || modeChanged) {
+            if (mode === SceneMode.SCENE3D) {
+                Cartesian3.clone(Cartesian3.ZERO, scratchRay.origin);
+                Cartesian3.normalize(object.position, scratchRay.direction);
+            } else {
+                ellipsoid.cartesianToCartographic(object.position, scratchCartographic);
+                scratchCartographic.height = -1000.0; // TODO: get minimum height of entire terrain set
+                projection.project(scratchCartographic, scratchPosition);
+                Cartesian3.fromElements(scratchPosition.z, scratchPosition.x, scratchPosition.y, scratchPosition);
+                Cartesian3.clone(scratchPosition, scratchRay.origin);
+                Cartesian3.clone(Cartesian3.UNIT_X, scratchRay.direction);
+            }
+
+            var position = tile.data.pick(scratchRay, mode, projection, false, scratchPosition);
```

Comment:
Can you submit a roadmap issue for billboards/labels on terrain that includes a checkbox to optimize the ray pick with a spatial data structure / adjacency info / temporal coherence / etc.?

Assume the performance is acceptable now, we will merge without this optimization.